source/slang/slang-check-overload.cpp

```diff
@@ -2241,6 +2246,33 @@ namespace Slang
         return argsListBuilder.produceString();
     }
 
+
+    /// We allow a special case for when `funcExpr` is expected to be a Default initializer
+    /// but none exist. Note, we cannot just create a default initializer for every variable
+    /// since then we are introducing initialization to every variable through an indirect
+    /// init returning data. Instead we will call `$ZeroInit` through this logic below.
+    Expr* _tryToSpecialCaseOverloadDefaultConstructWithoutInit(SemanticsVisitor* visitor, SemanticsVisitor::OverloadResolveContext& context, Expr* expr, OverloadCandidate* bestCandidate)
+    {
+        if (context.argCount == 0)
```

Comment:
should we make sure the canddiate we are calling is a ctor decl?

what are we trying to do here? If we have a ctor candidate that takes more than 0 parameters but we are trying to call it with 0 args, we form an initializerList coercion instead? Why do we need to do this?

If we have `struct S { __init (x) }` and called with `S s = {};`, it should be an error?

Reply:
This was a workaround we kept from the previous implementation (but changed to be less of a hack). The main idea is that some structs may be treated as if they have an __init despite not ever declaring an __init

The goal is that we will zeroInit these objects

Reply:
rename this function to `_tryOverloadWithDefaultInit`, and add some comment to explain this.
https://github.com/shader-slang/slang/pull/5107/commits/2b6f7a26688b60a74fde979fbfff28a24a493e5c